x-pack/plugins/security_solution/public/entity_analytics/pages/entity_analytics_management_page.tsx

```diff
@@ -5,38 +5,174 @@
+    }
+  };
+
+  const handleIncludeClosedAlertsToggle = (value: boolean) => {
+    setIncludeClosedAlerts(value);
+  };
+
+  const handleDateChange = ({ start, end }: { start: string; end: string }) => {
+    setFrom(start);
+    setTo(end);
+    localStorage.setItem('dateStart', start);
+    localStorage.setItem('dateEnd', end);
+  };
+
+  const calculateNextRunTime = useCallback((): string => {
+    if (runEngineEnabled) {
+      const currentTime = new Date();
+      const runAtTime = riskEngineStatus?.risk_engine_task_status?.runAt
+        ? new Date(riskEngineStatus.risk_engine_task_status.runAt)
+        : new Date();
+      const minutesUntilNextRun = Math.round((runAtTime.getTime() - currentTime.getTime()) / 60000);
+      return `Next engine run in ${minutesUntilNextRun} minutes`;
+    }
+    return '';
+  }, [runEngineEnabled, riskEngineStatus]);
+
+  useEffect(() => {
+    let intervalId: string | number | NodeJS.Timeout;
+
+    // Poll only if the engine is in the 'running' state
+    if (riskEngineStatus?.risk_engine_task_status?.status === 'idle') {
+      setNextRunTime(calculateNextRunTime()); // Initial calculation
+
+      intervalId = setInterval(() => {
+        setNextRunTime(calculateNextRunTime());
+      }, 60000); // Poll every minute
+    }
+
+    return () => {
+      if (intervalId) {
+        clearInterval(intervalId);
+      }
+    };
+  }, [calculateNextRunTime, riskEngineStatus?.risk_engine_task_status?.status]);
+
   return (
     <>
       <RiskEnginePrivilegesCallOut privileges={privileges} />
       <EuiPageHeader
         pageTitle={
-          <EuiFlexGroup>
+          <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
             <EuiFlexItem data-test-subj="entityAnalyticsManagementPageTitle" grow={false}>
               {ENTITY_ANALYTICS_RISK_SCORE}
             </EuiFlexItem>
-            <EuiBetaBadge label={BETA} size="s" />
+
+            <EuiFlexItem grow={false}>
+              <EuiFlexGroup alignItems="center" gutterSize="s">
+                {/* Run Engine Button */}
+                {runEngineEnabled && (
+                  <EuiFlexItem grow={false}>
+                    <EuiButton
+                      size="s"
+                      iconType="play"
+                      isLoading={isLoading}
+                      onClick={async () => {
+                        await handleRunEngineClick();
+                        if (!isLoading) {
+                          addSuccess(i18n.RISK_SCORE_ENGINE_RUN_SUCCESS, { toastLifeTimeMs: 5000 });
+                        }
+                      }}
+                    >
+                      {'Run Engine'}
+                    </EuiButton>
+                  </EuiFlexItem>
+                )}
+
+                {/* Vertical Line */}
+                {runEngineEnabled && (
+                  <div
+                    className="vertical-line"
+                    style={{ height: '24px', borderLeft: '1px solid #ccc', margin: '0 8px' }}
```

Comment:
Please use `theme` values instead of pixels. 
Please avoid using `style`

Reply:
Thanks, This is now being handled in `emotion` as i could not find a relative EUI component for a small vertical separator. The new changes are in the latest commit